### PR TITLE
Fix depwarn on latest master

### PR DESCRIPTION
**Pull request by eveydee**
Split it into two commits, since I'm not sure if the version check should be there or not with version tagging, etc.

### CODE REVIEW

File: src/Revise.jl [hunk 1]
```diff
@@ -319,7 +319,11 @@ function parse_source!(md::ModDict, file::AbstractString, mod::Module, path)
         warn("omitting ", file, " from revision tracking")
         return false
     end
-    parse_source!(md, readstring(file), Symbol(file), 1, mod, path)
+    if VERSION >= v"0.7.0-DEV"
```
Comment (timholy) [inline src/Revise.jl]: Thanks!

It's likely you'll have to be more specific about this version number (just using "DEV" has generated complaints in the past). If you can find the commit sha at which this change occurred (check the julia git log), then the `contrib/commit-name.sh` script in julia should give you the version number.

Comment (ararslan) [inline src/Revise.jl]: Alternatively, you could use the Compat package, which provides a `read(x, String)` method that calls `readstring` on older versions.

Comment (eveydee) [inline src/Revise.jl]: I can't find the `read` method in the latest release of the `Compat` package 🤷‍. And `@macroexpand @compat read(x, String)` returns `:(read(x, String))` on both 0.6 and master. I found the commit, though, so will update the version check.

Comment (ararslan) [inline src/Revise.jl]: Sorry, my bad—the Compat PR that includes `read(x, String)` hasn't been merged yet.

**Comment by timholy** [comment]
Thanks!

**Pull request merged**